cargo-cyclonedx/src/generator.rs

```diff
@@ -182,60 +182,63 @@ impl SbomGenerator {
         let mut subcomponents: Vec<Component> = Vec::new();
         let mut subcomp_count: u32 = 0;
         for tgt in &package.targets {
+            // Classification
             // Ignore tests, benches, examples and build scripts.
             // They are not part of the final build artifacts, which is what we are after.
-            if !(tgt.is_bench() || tgt.is_example() || tgt.is_test() || tgt.is_custom_build()) {
```

Comment:
I missed this during the review, but removing this line was wrong. 

We do need to filter out these things, and a crate will have more than one type at a time.

I'll have to revert this.

Reply:
I don't think that's correct.
I do check for that by checking for the reverse.
I only "include" lib and bin and continue otherwise.

Reply:
```
let cdx_type = match (tgt.is_bin(), tgt.is_lib()) {
                (true, false) => Classification::Application,
                (false, true) => Classification::Library,
                _ => {
                    log::warn!(
                        "Target {} is neither a binary nor a library! Kinds: {}",
                        tgt.name,
                        tgt.kind.join(", ")
                    );

                    continue;
                } // Skip if neither a binary nor a library
            };
```

Anything that is not `is_bin` or `is_lib` will be logged and skipped, just as before.
This should be more resilient should new kinds be added later.

Am I missing something?

Reply:
Yes, you are. It's an array of kinds, not a single specific kind. So `.is_example()` and `.is_bin()` are often true the same time.